WooCommerce/Classes/AppDelegate.swift

```diff
@@ -401,3 +415,89 @@ extension AppDelegate {
         RequirementsChecker.checkMinimumWooVersionForDefaultStore()
     }
 }
+
+// MARK: - Authentication helpers
+//
+private extension AppDelegate {
+    func isLoggedIn() -> Bool {
+        ServiceLocator.stores.isAuthenticated
+    }
+
+    func logout() {
+        ServiceLocator.stores.deauthenticate()
+    }
+}
+
+// MARK: - Apple Account Handling
+//
+@available(iOS 13.0, *)
+private extension AppDelegate {
```

Comment:
⚠️  I'm not sure but based on the code that I see here, we can encapsulate all of these into its own `class` to keep the `AppDelegate` lean-ish. We can _retain_ the object as a property of `AppDelegate` to keep it alive. 

We can use the [`didBecomeActive`](https://developer.apple.com/documentation/uikit/uiapplication/1622953-didbecomeactivenotification) `NSNotification` instead of relying on `AppDelegate`. And we could probably make that new `class` testable too. 

What do you think?

Reply:
I like the idea of encapsulating the Apple ID checks to its own class, rather than expanding the global `AppDelegate`! I created a class `AppleIDCredentialChecker` in https://github.com/woocommerce/woocommerce-ios/pull/2818/commits/55d07b2ad68c438944edd9b51bb31b68cf1ad363 and then updated `WordPressAuthenticator` to a lazy var in https://github.com/woocommerce/woocommerce-ios/pull/2818/commits/ad1786ee0ad4422bfc3887511af0715944cdff5c since the shared instance is only initialized in app delegate's `willFinishLaunchingWithOptions` after its first call in `applicationDidBecomeActive` 😅  (otherwise it crashes the app on launch)

For the check in `applicationDidBecomeActive`, I still let the app delegate call `AppleIDCredentialChecker`'s `checkAppleIDCredentialState`. But lemme know if you think observing the `NSNotification` within the class would be better!

Reply:
yea it makes sense to observe `applicationDidBecomeActive` within the class itself! updated in https://github.com/woocommerce/woocommerce-ios/pull/2818/commits/01bce49859e57d5592a8027e82bac0d46be49b87 and there's no need to make the authenticator a lazy var anymore (https://github.com/woocommerce/woocommerce-ios/pull/2818#discussion_r490672218)!